app/javascript/crayons/MentionAutocomplete/MentionAutocompleteCombobox.jsx

```diff
@@ -0,0 +1,169 @@
+  );
+};
+
+/**
+ * A component for dynamically searching for users and displaying results in a dropdown.
+ * This component should be mounted when a user has started typing a mention with the '@' symbol, and will be positioned at the given coordinates.
+ *
+ * @param {object} props
+ * @param {string} props.startText The initial search term to use
+ * @param {function} props.onSelect Callback function for using the selected user
+ * @param {function} props.fetchSuggestions The async call to use for the search
+ * @param {object} props.placementCoords The x/y coordinates for placement of the popover
+ * @param {function} props.onSearchTermChange A callback for each time the searchTerm changes
+ *
+ * @example
+ * <MentionAutocompleteCombobox
+ *    startText="name"
+ *    onSelect={handleUserMentionSelection}
+ *    fetchSuggestions={fetchUsersByUsername}
+ *    placementCoords={{x: 22, y: 0}}
+ *    onSearchTermChange={updateSearchTermText}
+ * />
+ */
+export const MentionAutocompleteCombobox = ({
+  onSelect,
+  fetchSuggestions,
+  placementCoords,
+  onSearchTermChange,
+}) => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const [cachedSearches, setCachedSearches] = useState({});
+  const [users, setUsers] = useState([]);
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    if (searchTerm.trim() !== '' && searchTerm.length >= 2) {
+      if (cachedSearches[searchTerm]) {
+        setUsers(cachedSearches[searchTerm]);
+        return;
+      }
+
+      fetchSuggestions(searchTerm).then((fetchedUsers) => {
+        setCachedSearches({ ...cachedSearches, [searchTerm]: fetchedUsers });
+        setUsers(fetchedUsers);
+      });
+    }
+  }, [searchTerm, fetchSuggestions, cachedSearches]);
+
+  useEffect(() => {
+    inputRef.current.focus();
+  }, [inputRef]);
+
+  useLayoutEffect(() => {
+    const popover = document.getElementById('mention-autocomplete-popover');
+    if (!popover) {
+      return;
+    }
+
+    const closeOnClickOutsideListener = (event) => {
+      if (!popover.contains(event.target)) {
+        // User clicked outside, exit with current search term
+        onSelect(searchTerm);
+      }
+    };
+
+    document.addEventListener('click', closeOnClickOutsideListener);
+
+    return () =>
+      document.removeEventListener('click', closeOnClickOutsideListener);
+  }, [searchTerm, onSelect]);
+
+  const handleSearchTermChange = (event) => {
+    const {
+      target: { value },
+    } = event;
+
+    if (value === '' || value.charAt(value.length - 1) === ' ') {
+      // User has deleted their selection or spaced away from a complete word - finish the autocomplete
+      onSelect(value);
+      return;
+    }
+    setSearchTerm(value);
+    onSearchTermChange(value);
+  };
+
+  return (
+    <Combobox
+      aria-label="mention user"
+      onSelect={(item) => onSelect(item)}
+      className="crayons-autocomplete"
+    >
+      <ComboboxInput
+        style={{
+          opacity: 0.000001,
```

Comment:
We want this hidden from view, but we need a v small amount of opacity otherwise ChromeVox screen reader will ignore it